apps/src/studio/Sprite.js

```diff
@@ -0,0 +1,459 @@
+  // call collidable constructor
+  Collidable.apply(this, arguments);
+
+  /**
+   * Rendering offset for item animation vs display position - applied as
+   * late as possible.
+   * @type {{x: number, y: number}}
+   */
+  this.renderOffset = options.renderOffset || { x: 0, y: 0 };
+
+  this.speed = options.speed || constants.DEFAULT_SPRITE_SPEED;
+  this.displayDir = Direction.NONE;
+  this.startFadeTime = null;
+  this.fadeTime = 0;
+
+  this.image = null;
+  this.legacyImage = null;
+
+  this.animation_ = null;
+  this.legacyAnimation_ = null;
+
+  this.useLegacyIdleEmotionAnimations = false;
+
+  this.lastDrawPosition = { x: 0, y: 0 };
+};
+Sprite.inherits(Collidable);
+module.exports = Sprite;
+
+/**
+ * frameCounts objects are used in setImage and setLegacyImage:
+ *
+ *  walkFrames, idleFrames: frames in each type of animations (spriteSheet)
+ *  walk: default frames in all types of animations (spriteSheet)
+ *  idleNormal, idleEmotions, walkingEmotions: count of each type of animation
+ *    (applies to spriteSheet)
+ *  counterClockwise (boolean): (applies to spriteSheet)
+ *  turns: how many turn animations (applies to spriteSheet && legacySpriteSheet)
+ *  normal, emotions, extraEmotions: (applies to legacySpriteSheet)
+ *  packedSheetFrameCount: count of frames before wrapping for a packed sheet
+ *    (applies to spriteSheet)
+ */
+
+/**
+ * Sets (or modifies) the image - we will generate a StudioSpriteSheet and
+ * StudioAnimation in response..
+ */
+Sprite.prototype.setImage = function (image, frameCounts) {
+  if (image !== this.image) {
+    this.image = image;
+
+    var options = {
+      renderScale: this.renderScale,
+      opacity: this.opacity,
+      loop: this.loop,
+      animationFrameDuration: this.animationFrameDuration,
+      frameWidth: this.drawWidth,
+      frameHeight: this.drawHeight,
+      assetPath: this.image,
+      animations: [
+        {
+          type: 'direction',
+          count: frameCounts.turns || 0,
+          frames: frameCounts.walkFrames || frameCounts.walk
+        },
+        {
+          type: 'idle',
+          count: (frameCounts.idleNormal || 0) + (frameCounts.idleEmotions || 0),
+          frames: frameCounts.idleFrames || frameCounts.walk
+        },
+        {
+          type: 'walkingEmotions',
+          count: frameCounts.walkingEmotions || 0,
+          frames: frameCounts.walkFrames || frameCounts.walk
+        }
+      ],
+      packedSheetFrameCount: frameCounts.packedSheetFrameCount,
+      defaultFramesPerAnimation: frameCounts.walk,
+      skewAnimations: true
+    };
+
+    if (this.animation_) {
+      this.animation_.removeElement();
+    }
+
+    /** @private {StudioAnimation} */
+    this.animation_ = new StudioAnimation($.extend({}, options, {
+      spriteSheet: new StudioSpriteSheet(options),
+      animationFrameDuration: this.getAnimationFrameDuration()
+    }));
+  }
+};
+
+Sprite.prototype.setLegacyImage = function (image, frameCounts) {
```

Comment:
This function would be a good place to document what we mean by "Legacy" image, or to point to some other place where that is explained.